### PR TITLE
Fix Clang w/ MinGW build configuration

### PR DESCRIPTION
**Pull request by Keno**
Has to be Clang 3.8.0, but with these changes, everything seems to be essentially working.

### CODE REVIEW

File: src/gc.c [hunk 1]
```diff
@@ -181,7 +181,7 @@ typedef struct {
     uint32_t freemap[REGION_PG_COUNT/32];
     gcpage_t meta[REGION_PG_COUNT];
 } region_t
-#ifndef _COMPILER_MICROSOFT_
+#if !defined(_COMPILER_MICROSOFT_) && !(defined(_COMPILER_MINGW_) && defined(_COMPILER_CLANG_))
```
Comment (tkelman) [inline src/gc.c]: clang-on-mingw doesn't handle `__attribute__((aligned))` ?

Comment (Keno) [inline src/gc.c]: COFF is limited to 8192 alignments.

Comment (tkelman) [inline src/gc.c]: is mingw ignoring then, or saturating?

Comment (Keno) [inline src/gc.c]: If I had to guess, I'd say overflowing ;), but I don't know.

**Comment by tkelman** [comment]
So is 3.8 the first clang version with complete enough seh support to build fully working versions of itself and our other c++ deps? What about on 32 bit?

**Comment by Keno** [comment]
I have not tried 32 bit.

**Pull request merged**